src/app/parts-editor/components/dialog-database/dialog-database.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { FormBuilder, FormGroup, FormArray } from '@angular/forms';
+import { PartWorkbook } from '../../models/editor';
+import { Subject } from 'rxjs';
+import { Cell } from '../../models/cell';
+
+@Component({
+  selector: 'app-dialog-database',
+  templateUrl: './dialog-database.component.html',
+  styleUrls: ['./dialog-database.component.scss'],
+})
+export class DialogDatabaseComponent implements OnInit, OnDestroy {
+  destroy$ = new Subject<void>();
+  missingPartsForm: FormGroup;
+
+  constructor(
+    public dialogRef: MatDialogRef<DialogDatabaseComponent>,
+    private formBuilder: FormBuilder,
+    @Inject(MAT_DIALOG_DATA) public missingParts: PartWorkbook
+  ) {}
+
+  ngOnInit() {
+    this.missingPartsForm = this.formBuilder.group({
+      updatedParts: this.formBuilder.array([]),
+    });
+
+    for (const part of this.missingParts) {
+      this.addPart(part);
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  updatedParts(): FormArray {
+    return this.missingPartsForm.get('updatedParts') as FormArray;
+  }
+
+  addPart(part: {[key: string]: Cell}) {
+    const newPart = this.formBuilder.group({
+      rowIndex: part.articleNo.row,
+      nameReport: part.articleNo.value,
+      nameSap: '',
+      numberSap:'',
+      addToDatabase: false
+    });
+
+    this.updatedParts().push(newPart);
+  }
+
+  newPart(partNumber: string): FormGroup {
```

Comment:
Function name should describe the action its doing. Do not use nouns for function names unless its a getter / setter and it mimics the property.  Here I suggest using createPartForm()